File: Vindeep-Chaudhari/1863_Sum_of_All_Subset_XOR_Totals.cpp
/**
 * Problem Link - https://leetcode.com/problems/sum-of-all-subset-xor-totals/
 */
class Solution {
public:
    void getSubsetsAndSum(vector<int>& nums, int& XorSum, int i, int xorTill) {
        if(i == nums.size()) return;

         XorSum += xorTill^nums[i];

        getSubsetsAndSum(nums, XorSum, i+1, xorTill^nums[i]);
        getSubsetsAndSum(nums, XorSum, i+1, xorTill);
    }
    int subsetXORSum(vector<int>& nums) {

        int XorSum = 0;
        getSubsetsAndSum(nums, XorSum, 0, 0);

        return XorSum;
    }
};

File: Namrata-Sanger/1863. Sum of All Subset XOR Totals.js
/*
Day 37
1863. Sum of All Subset XOR Totals
Link: https://leetcode.com/problems/sum-of-all-subset-xor-totals/submissions/
Level: Easy 
*/

/**
 * @param {number[]} nums
 * @return {number}
 */
var subsetXORSum = function(nums) {
    let output=[];
    backtrack();
    return output.reduce((a,b)=>a+b);
    function backtrack(start = 0, arr=[nums[0]]){
       output.push([...arr].reduce((a,b)=>a^b,0));
       for(let i=start; i<nums.length; i++){
            arr.push(nums[i]);
            backtrack(i+1, arr);
            arr.pop();
       }
    }
};
